src/strategies/pancake/index.ts

```diff
@@ -0,0 +1,64 @@
+import { formatUnits } from "@ethersproject/units";
+import { multicall } from "../../utils";
+import { strategy as erc20BalanceOfStrategy } from "../erc20-balance-of";
+
+export const author = "pancake";
+export const version = "0.0.1";
+
+const abi = [
+  {
+    inputs: [
+      {
+        internalType: "address",
+        name: "",
+        type: "address",
+      },
+    ],
+    name: "userInfo",
+    outputs: [
+      {
+        internalType: "uint256",
+        name: "amount",
+        type: "uint256",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  },
+];
+
+export async function strategy(network, provider, addresses, options, snapshot) {
+  const blockTag = typeof snapshot === "number" ? snapshot : "latest";
+  const score = await erc20BalanceOfStrategy(network, provider, addresses, options, snapshot);
+
+  const balances = await Promise.all(
+    options.chefAddresses.map((chefAddress) =>
```

Comment:
I'm concerned that this will send too much request at the same time and overload user browsers, is there a way you can call the contract with only 1 to 3 requests?

Reply:
chefAddresses will be only 2-4 length